jstz_cli/src/account/mod.rs

```diff
@@ -102,29 +124,72 @@ pub fn whoami(cfg: &mut Config) -> Result<()> {
         .as_ref()
         .ok_or(anyhow!("Not logged in!"))?;
 
-    let account = cfg.accounts.get(&alias)?;
+    let (alias, address) =
+        if let Account::Owned(OwnedAccount { alias, address, .. }) =
+            cfg.accounts.get(&alias)?
+        {
+            (alias, address)
+        } else {
+            return Err(anyhow!("Account is not owned"));
+        };
 
-    println!(
-        "Logged in to account {} with address {}",
-        account.alias, account.address
-    );
+    println!("Logged in to account {} with address {}", alias, address);
 
     Ok(())
 }
 
-fn list(cfg: &mut Config) -> Result<()> {
+fn list_accounts(cfg: &mut Config, long: bool) -> Result<()> {
     let accounts = cfg.accounts().list_all();
 
     println!("Accounts:");
     for (alias, account) in accounts {
-        println!("{}: {}", alias, account.address);
+        if long {
+            println!("Alias: {}", alias);
+            match account {
+                Account::Owned(OwnedAccount {
+                    alias: _,
+                    nonce,
+                    address,
+                    secret_key,
+                    public_key,
+                }) => {
+                    println!("  Type: Owned");
+                    println!("  Nonce: {:?}", nonce);
+                    println!("  Address: {}", address);
+                    println!("  Public Key: {}", public_key.to_string());
+                    println!("  Secret Key: {}", secret_key.to_string());
+                }
+                Account::Alias(AliasAccount { address, .. }) => {
+                    println!("  Type: Alias");
+                    println!("  Address: {}", address);
+                }
+            }
+        } else {
+            match account {
+                Account::Owned(OwnedAccount { alias, address, .. }) => {
+                    println!("{}: {}", alias, address)
+                }
+                Account::Alias(AliasAccount { alias, address, .. }) => {
+                    println!("{}: {}", alias, address)
+                }
+            }
+        }
```

Comment:
Given it is so common to get `alias` and `address` from an `Account`, might be worth adding `alias()` and `address()` methods to `Account`.